### DISCUSSION TITLE
What Databases can I use?

### DISCUSSION THREAD

**Discussion by MastersoftDev** [pascalabcnet/pascalabcnet#2712] [Q&A] [answered]
Hello friends,
Thank you for this wonderful Pascal-Based IDE. I would be so much interested to use it but I would like to know the types of Databases that can be used in this IDE. I develop desktop applications using databases. Does SQLite work? What about MySql?

Thank you
Adrian

**Comment by samuraiGH** [comment, marked as answer]
Hello
> Thank you for this wonderful Pascal-Based IDE

It is not only an IDE, but also a powerful compiler that introduces many extensions to classic Pascal

> Does SQLite work?

Yes, you should download the appropriate version of System.Data.SQLite [here](https://system.data.sqlite.org/index.html/doc/trunk/www/downloads.wiki)
Minimal code with this library
```pas
{$reference System.Data.SQLite.dll}

## uses System.Data.SQLite;

var con:= new SQLiteConnection('Data Source=Application.db;');
con.Open;
var command:= new SQLiteCommand(
  'CREATE TABLE users (id INT)',
  con
);
command.ExecuteNonQuery;

con.Close;
```

**Reply by MastersoftDev** [reply]
Dear SamuraiGH,
Thank you for the reply. I've highly appreciated your support.
I would like to let you know that I'm new to Pascal Language though I'm very eager to learn and to use it in software development. 
(1). Where can I get a good tutorial to either read or watch to get my hands on quickly?
(2). Can you please also how I can load data from SQlite Database into the Datagridview?

Thank You

Adrian

**Reply by samuraiGH** [reply]
> Where can I get a good tutorial to either read or watch to get my hands on quickly?

Unfortunately, I do not know the literature specifically on PascalABC.NET it is in English. According to classical Pascal, such literature is probably easy to find. But the old Pascal and PascalABC.NET they differ quite a lot.

You can use Russian literature if it is acceptable to you. 
A good material is the help built into the IDE. 
There are more books in Russian on the [site](http://pascalabc.net/books).
There is also a [course on stepik](https://stepik.org/course/91781/promo).

> Can you please also how I can load data from SQlite Database into the Datagridview?

There are quite a few options to do this. If you are familiar with `System.Windows.Forms.dll` and/or` System.Data.dll`, then tell me what is better to use for an example. If you are not familiar with these libraries, then the example is likely to be quite complicated for a beginner in the language

**Reply by MastersoftDev** [reply]
Thank you very much for your response. I've highly appreciated your timely support to me.
Concerning the issue of loading data from SQlite Database, I've been able to do so with the following code;

procedure KJVForm.button1_Click(sender: Object; e: EventArgs);
begin
  var connection := new SQLiteConnection('Data Source=D:\Daniel\MasterDb.db;');
  connection.Open;
  var command := new SQLiteCommand('SELECT * FROM t_akjv WHERE BookNo=@UniqueVerseID', connection);
  command.Parameters.Add('@UniqueVerseID', DbType.String).Value := Convert.ToString('1');
  var adapter := new SQLiteDataAdapter(command);
  var dTable := new DataTable;
  adapter.Fill(dTable);
  dataGridView1.DataSource := dTable;  
  connection.Close;
end;

I'm I on the right direction?
But still, the help file is in Russian language which I do not know!!! I still need much help in this regard.

Thanks

Adrian

**Reply by samuraiGH** [reply]
> I'm I on the right direction?

Yes. This is one of the possible options.
> But still, the help file is in Russian language which I do not know!!!

To date, translators are sufficiently developed for high-quality translation. My messages are also translated from Russian automatically and are only slightly corrected by me. I understand that this is not very convenient for learning, but I can't offer anything better. Unfortunately, the training materials in English have not yet been created

**Reply by MastersoftDev** [reply]
Dear SamuraiGH,
Thank you very much for your support.
So far, I'm progressing on well. But do you have a plan of providing us with a "Showcase" space on the main website where we can show the software Applications that we've made with PascalABCNET?

Thank you

Adrian

**Reply by MastersoftDev** [reply]
One more last point, can help me know how I can get the DataGridView Cell Value into a TextBox? It seems hard for me to do it. In Visual Basic.net I do it like this;

		If e.RowIndex >= 0 Then
			Dim row As DataGridViewRow = purchaseItemsGridView.Rows(e.RowIndex)
			If row.Index = purchaseItemsGridView.Rows.Count - 1 Then
				Return
			End If
			txtItemNo.Text = (row.Cells(0).Value.ToString)
			txtItemName.Text = (row.Cells(2).Value.ToString)
			txtQuantity.Text = (row.Cells(3).Value.ToString)
			txtUnitCost.Text = (row.Cells(4).Value.ToString)
			txtAmount.Text = (row.Cells(5).Value.ToString)
		End If
I'm failing to write the same code in PascalABCNET! Please, can you help me?

Thank you,

Adrian

**Reply by SunSerega** [reply]
@samuraiGH isn't a dev of this repo (and neither am I), so you asked the wrong person about "providing".

Usually, the [forum](http://forum.mmcs.sfedu.ru/c/pascalabcnet/10) is used as a showcase space, in particular [this thread](http://forum.mmcs.sfedu.ru/t/proekty-na-pascalabc-net/2272/210).

Though it's also all in Russian, so far there weren't many English-only users of this language.

**Reply by SunSerega** [reply]
> I'm failing to write the same code in PascalABCNET!

What exactly did you fail to translate?

**Reply by MastersoftDev** [reply]
One more last point, can help me know how I can get the DataGridView Cell Value into a TextBox? It seems hard for me to do it. In Visual Basic.net I do it like this;

	If e.RowIndex >= 0 Then
		Dim row As DataGridViewRow = purchaseItemsGridView.Rows(e.RowIndex)
		If row.Index = purchaseItemsGridView.Rows.Count - 1 Then
			Return
		End If
		txtItemNo.Text = (row.Cells(0).Value.ToString)
		txtItemName.Text = (row.Cells(2).Value.ToString)
		txtQuantity.Text = (row.Cells(3).Value.ToString)
		txtUnitCost.Text = (row.Cells(4).Value.ToString)
		txtAmount.Text = (row.Cells(5).Value.ToString)
	End If
I'm failing to write the same code in PascalABCNET! Please, can you help me?

Thank you,

Adrian
